src/toolkits/style_transfer/style_transfer.cpp

```diff
@@ -188,6 +236,125 @@ void style_transfer::infer_derived_options() {
   add_or_update_state({{"training_iterations", 0}});
 }
 
+gl_sframe style_transfer::predict(variant_type data,
+                                  std::map<std::string, flexible_type> opts) {
+  gl_sframe_writer result({"style_idx", "stylized_image"},
+                          {flex_type_enum::INTEGER, flex_type_enum::IMAGE}, 1);
+
+  gl_sarray content_images = convert_types_to_sarray(data);
+
+  std::vector<double> style_idx;
+  auto style_idx_iter = opts.find("style_idx");
+  if (style_idx_iter == opts.end()) {
+    size_t num_styles = read_state<flexible_type>("num_styles");
+    style_idx.resize(num_styles);
+    std::iota(style_idx.begin(), style_idx.end(), 0);
+  } else {
+    flexible_type flex_style_idx = style_idx_iter->second;
+    switch (flex_style_idx.get_type()) {
+      case flex_type_enum::INTEGER:
+        style_idx.push_back(flex_style_idx.get<flex_int>());
+        break;
+      case flex_type_enum::VECTOR:
+        style_idx = std::move(flex_style_idx.get<flex_vec>());
+        break;
+      default:
+        log_and_throw(
+            "Option \"style_idx\" has to be of type `Integer` or `SArray`.");
+    }
+  }
+
+  perform_predict(content_images, result, style_idx);
+
+  return result.close();
+}
+
+void style_transfer::perform_predict(gl_sarray data, gl_sframe_writer& result,
+                                     const std::vector<double>& style_idx) {
+  if (data.size() == 0) return;
+
+  flex_int batch_size = read_state<flex_int>("batch_size");
+  flex_int num_styles = read_state<flex_int>("num_styles");
+  flex_int image_width = read_state<flex_int>("image_width");
+  flex_int image_height = read_state<flex_int>("image_height");
+
+  // Since we aren't training the style_images are irrelevant
+  std::unique_ptr<data_iterator> data_iter = create_iterator(
+      data, /* style_sframe */ {}, /* repeat */ false, num_styles);
+
+  std::unique_ptr<compute_context> ctx = create_compute_context();
+  if (ctx == nullptr) {
+    log_and_throw("No neural network compute context provided");
+  }
+
+  float_array_map weight_params = m_resnet_spec->export_params_view();
+
+  // A value of `0` to indicate prediction
+  shared_float_array st_train(std::make_shared<float_scalar>(0));
```

Comment:
FYI you also write this as:
```
shared_float_array st_train = shared_float_array::wrap(0.f);
```
using this factory function: https://github.com/apple/turicreate/blob/65ef3e972953776d186c8d6a12f3e23c64a4d55b/src/ml/neural_net/float_array.hpp#L140